lib/elixir/test/elixir/kernel/binary_test.exs

```diff
@@ -269,6 +269,16 @@ defmodule Kernel.BinaryTest do
     assert <<1::size((^foo).bar)>> = <<1::5>>
   end
 
+  test "automatic size computation of matched bitsyntax variable" do
+    var = "foo"
+    <<^var::binary, rest::binary>> = "foobar"
```

Comment:
it would be really nice if this worked too:

```elixir
^var <> rest = "foobar"
```

Reply:
It will work, <> expands to the above.

Reply:
Needs to be addressed though, right now it fails with
```
** (ArgumentError) the left argument of <> operator inside a match should always be a literal binary because its size can't be verified. Got: ^var
```

Reply:
Ah, we probably explicit forbid it for better error messages.

Reply:
Let's do it here, I think it will be mostly removing code.

Reply:
It was indeed 🙂
[ff24c64](https://github.com/elixir-lang/elixir/pull/13106/commits/ff24c64e6f4a5db66243710e464dae37c1ee9d5d)